getting-started/quickstart-guides/java-guides/java_app.md

```diff
@@ -0,0 +1,334 @@
+
+ImageKit.io allows you to add [text](../../../features/image-transformations/overlay.md#text-overlay) and [image overlay](../../../features/image-transformations/overlay.md) dynamically.
+
+#### **Text overlay**
+
+Text overlay can be used to superimpose text on an image. For example:
+
+```java
+List<Map<String, String>> transformation=new ArrayList<Map<String, String>>();
+Map<String, String> scale=new HashMap<>();
+scale.put("height","300");
+scale.put("width","300");
+transformation.add(scale);
+Map<String, String> overlay=new HashMap<>();
+overlay.put("overlayText","ImageKit");
+overlay.put("overlayTextFontSize","50");
+overlay.put("overlayTextColor","0651D5");
+transformation.add(overlay);
+Map<String, Object> options=new HashMap();
+options.put("path", "/default-image.jpg");
+options.put("transformation", transformation);
+
+String image_url=ImageKit.getInstance().getUrl(options);
+```
+
+**Transformation URL:**
+
+```http
+https://ik.imagekit.io/zv3rkhsym/tr:w-300,h-300:ots-50,ot-ImageKit,otc-0651D5/default-image.jpg?ik-sdk-version=java-1.0.3
+```
+
+**Output Image:**
+
+![Text Overlay (300px \* 300px)](<../../../.gitbook/assets/java-app-with-overlay-text.png>)
+
+#### **Image overlay**
+
+Image overlay can be used to superimpose an image on another image. For example, we will upload a while logo image on [this link](https://ik.imagekit.io/demo/logo-white_SJwqB4Nfe.png) into our account and use it for the overlay image.
+
+Base Image: `default-image.jpg`
+
+Overlay Image: `logo-white_SJwqB4Nfe.png`
+
+```java
+List<Map<String, String>> transformation=new ArrayList<Map<String, String>>();
+Map<String, String> scale=new HashMap<>();
+scale.put("height","300");
+scale.put("width","300");
+transformation.add(scale);
+Map<String, String> overlay=new HashMap<>();
+overlay.put("overlayImage","logo-white_SJwqB4Nfe.png");
+transformation.add(overlay);
+Map<String, Object> options=new HashMap();
+options.put("path", "/default-image.jpg");
+options.put("transformation", transformation);
+
+String image_url=ImageKit.getInstance().getUrl(options);
+```
+
+**Transformation URL:**
+
+```http
+https://ik.imagekit.io/zv3rkhsym/tr:w-300,h-300:oi-logo-white_SJwqB4Nfe.png/default-image.jpg?ik-sdk-version=java-1.0.3
+```
+
+**Output Image:**
+
+![Overlay image over another image](<../../../.gitbook/assets/java-app-with-overlay-image.png>)
+
+## Secure signed URL generation
+
+You can use the SDK to generate a signed URL of an image, that expires in a given number of seconds.
+
+```java
+Map<String, Object> options=new HashMap();
+options.put("path", "/default-image.jpg");
+options.put("signed",true);
+options.put("transformation", new ArrayList<Map<String, String>>());
+options.put("expireSeconds",10);
+
+String signed_image_url=ImageKit.getInstance().getUrl(options);
+```
+
+The above snippets create a signed URL with an expiry time of 10 seconds.
+
+![Signed URL generation](<../../../.gitbook/assets/java-app-with-signed-image.png>)
+
```

Comment:
Only URL generation is covered in the quick-start guide? What about the rest of the methods like we cover in https://docs.imagekit.io/getting-started/quickstart-guides/php (old). [New PR](https://github.com/imagekitio/docs-keep-a-changelog/pull/36)